pkg/scheduler/framework/session.go

```diff
@@ -297,6 +297,15 @@ func (ssn *Session) Allocate(task *api.TaskInfo, hostname string) error {
 				return err
 			}
 		}
+	} else {
+		// For best-effort pod, it it not necessary to consider the gang constraint.
+		if task.InitResreq.IsEmpty() {
```

Comment:
what's happen if there not enough resources for the other Pods?

Reply:
If there are six pods in a job, and the `minA` is `6`, two of the the six pods are best-effort pods, and the others are non-best-effort pods. The best-effort pods will be dispatched succeed and refreshed to `running` while the non-best-effort pods will be remain `pending` if there are not enough resources to satisfy the resoure requests of the job in the cluster.